src/jrd/version.h

```diff
@@ -0,0 +1,151 @@
+/*
+ *  The contents of this file are subject to the Initial
+ *  Developer's Public License Version 1.0 (the "License");
+ *  you may not use this file except in compliance with the
+ *  License. You may obtain a copy of the License at
+ *  http://www.ibphoenix.com/main.nfs?a=ibphoenix&page=ibp_idpl.
+ *
+ *  Software distributed under the License is distributed AS IS,
+ *  WITHOUT WARRANTY OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing rights
+ *  and limitations under the License.
+ *
+ *  The Original Code was created by Vladyslav Khorsun
+ *  for the Firebird Open Source RDBMS project.
+ *
+ *  Copyright (c) 2019 Vladyslav Khorsun <hvlad@users.sourceforge.net>
+ *  and all contributors signed below.
+ *
+ *  All Rights Reserved.
+ *  Contributor(s): ______________________________________.
+ */
+
+#ifndef JRD_VERSION_H
+#define JRD_VERSION_H
+
+#define STRINGIZE_AUX(x)	#x
+#define STRINGIZE(x)		STRINGIZE_AUX(x)
+
+
+#ifdef RC_TARGET_engine13
+#define VER_FILEDESC "Engine plugin"
+
+#elif defined RC_TARGET_fb_lock_print
+#define VER_FILEDESC "Lock Print tool"
+
+#elif defined RC_TARGET_fbguard
+#define VER_FILEDESC "Guardian"
+
+#elif defined RC_TARGET_fbrmclib
+#define VER_FILEDESC "RM/COBOL client interface"
+
+#elif defined RC_TARGET_firebird
+#define VER_FILEDESC "Server main executable"
+
+#elif defined RC_TARGET_fbsvcmgr
+#define VER_FILEDESC "Service Manager"
+
+#elif defined RC_TARGET_fbtrace
+#define VER_FILEDESC "Trace plugin"
+
+#elif defined RC_TARGET_fbtracemgr
+#define VER_FILEDESC "Trace Manager"
+
+#elif defined RC_TARGET_gbak
+#define VER_FILEDESC "Gbak tool"
+
+#elif defined RC_TARGET_gfix
+#define VER_FILEDESC "Gfix tool"
+
+#elif defined RC_TARGET_gpre
+#define VER_FILEDESC "Gpre tool"
+
+#elif defined RC_TARGET_gsec
+#define VER_FILEDESC "Gsec tool"
+
+#elif defined RC_TARGET_gsplit
+#define VER_FILEDESC "Gsplit tool"
+
+#elif defined RC_TARGET_gstat
+#define VER_FILEDESC "Gstat tool"
+
+#elif defined RC_TARGET_ib_util
+#define VER_FILEDESC "UDF support library"
+
+#elif defined RC_TARGET_instclient
+#define VER_FILEDESC "Install Client tool"
+
+#elif defined RC_TARGET_instreg
+#define VER_FILEDESC "Install Registry tool"
+
+#elif defined RC_TARGET_instsvc
+#define VER_FILEDESC "Install Service tool"
+
+#elif defined RC_TARGET_fbintl
+#define VER_FILEDESC "International Characters support library"
+
+#elif defined RC_TARGET_isql
+#define VER_FILEDESC "Interactive Query tool"
+
+#elif defined RC_TARGET_legacy_auth
+#define VER_FILEDESC "Legacy Auth plugin"
+
+#elif defined RC_TARGET_legacy_usermanager
+#define VER_FILEDESC "Legacy User Manager plugin"
+
+#elif defined RC_TARGET_nbackup
+#define VER_FILEDESC "Physical Backup Manager"
```

Comment:
Physical Backup tool?

Reply:
nbackup usage screen say : "Physical Backup Manager"

Reply:
We have "executable", "library", "plugin", "tool" -- they're well understood. What is "interface" and how it differs from "library"? What is "manager" and how it differs from "tool"? Why GBAK is "tool" but NBACKUP is "manager"? ;-) I know this is nitpicking and I surely accept the current names too. But perhaps we could be more consistent.